src/modules/Elsa.Workflows.Core/Contracts/IInstanceNameRetriever.cs

```diff
@@ -0,0 +1,12 @@
+namespace Elsa.Workflows.Contracts;
+
+/// <summary>
+/// Retrieves a name of the current instance.
+/// </summary>
+public interface IInstanceNameRetriever
```

Comment:
Although with enough context, "instance" should be fine, but perhaps to make it extra clear, let's rename this to `IApplicationInstanceNameAccessor`.